app/javascript/optimization/refresh.js

```diff
@@ -0,0 +1,21 @@
+export function refresh() {
+  if (refresh.url === null) {
+    console.error("optimize: refresh not initialized");
+    return;
+  }
+
+  window.miqSparkleOn();
+
+  return http.get(refresh.url)
+    .then((data) => {
+      refresh.set(data);
+
+      window.miqSparkleOff();
+      return data;
+    })
+    .catch(() => window.miqSparkleOff());
+}
+
+// overriden on OptimizationList init
+refresh.url = null;
```

Comment:
does this comes from `window?`

Reply:
What does?

`refresh` is the function defined right above it,
but until the react component itself is initialized, we don't know the request url (or the state setter).

Reply:
If it helps...

`refresh()` gets called whenever we receive the Rx message from toolbar, and after queuing a report rerun,
  it uses `refresh.url` to know where to refresh from,
  and `refresh.set` to know what to refresh.

`OptimizationList` sets `refresh.url` and `refresh.set` on load, from props and useState respectively.

Reply:
ah i see sorry